diff -Naru a/src/cpp/cuda.hpp b/src/cpp/cuda.hpp
--- a/src/cpp/cuda.hpp	2023-10-07 08:23:08.000000000 +0200
+++ b/src/cpp/cuda.hpp	2025-05-13 16:14:32.631476583 +0200
@@ -859,7 +859,12 @@
     context::prepare_context_switch();

     CUcontext ctx;
+#if CUDA_VERSION >= 13000
+    CUctxCreateParams params = {};
+    CUDAPP_CALL_GUARDED_THREADED(cuCtxCreate, (&ctx, &params, flags, m_device));
+#else
     CUDAPP_CALL_GUARDED_THREADED(cuCtxCreate, (&ctx, flags, m_device));
+#endif
     boost::shared_ptr<context> result(new context(ctx));
     context_stack::get().push(result);
     return result;
